include/spinnaker.h

```diff
@@ -1101,7 +1101,8 @@ enum {
 // On-chip SDRAM
 //! Bits in #GPIO_PORT, also on-chip SDRAM
 enum {
-    SDRAM_TQ =      (1 << 31),
+    // Out of range, so shouldn't be used
+    // SDRAM_TQ =      (1 << 31),
```

Comment:
It seems to me that this is __not__ the right way to declare these values. This value is __not__ out range and can be used when interfacing to the SDRAM.

These are bit positions, which are, by their nature, unsigned integers. The underlying type of C __enums__ is integer-_compatible_ but implementation defined. In this compiler, they are _signed_.

I wouldn't rule out a perfectly valid bit position declaration in order to be able to use a compiler construct.

Reply:
No, I guess that means we should go back to the #define version of this.  It is a shame that we can't force the compiler to use unsigned integers in enums, but it doesn't seem possible in this case.

Reply:
I agree that it's a shame and also that we should go back to the #define version.

Reply:
I think the warning that is generated when an enum is used here is "just" a warning according to the documentation and that an unsigned integer will in fact be used by this enum.  The shift was also causing some issues, so I have made a #define for this value.  Once that is in place, a little bit of armcc pragma magic stops the warning popping up everywhere!

Is that OK?